articles/introduction-to-constructors-in-c++/index.md

```diff
@@ -0,0 +1,270 @@
+### Introduction
+A constructor is a special member function that initializes the objects of its class. They are said to be special because their name is the same as the class name. Constructors are invoked every time an instance of the class is created. They automate the destroying of allocated variables therefore the programmer doesn’t have to create a function call to destroy them. This article will go through basics of C++ constructors.
+
+#### Prerequisites
+To follow through this article, you will need:
+  - Basic understanding of C++ language
+  - Basic understanding of C++ functions
+  - [Codeblocks](http://www.codeblocks.org/downloads) IDE to run the code.
+
+### What we’ll go through:
+  1. [Syntax of constructor functions](#syntax-of-constructor-functions)
+  2. [Characteristics of constructor functions](#characteristics-of-constructor-functions)
+  3. [Types of Constructors](#types-of-constructors)
+
+### Syntax of constructor functions
+```c++
+class integer
+{
+  int a , b;
+public:
+  integer(void); // constructor declaration
+};
+
+integer:: integer(void) // constructor definition
+{
+  a=2;
+  b=2;
+}
+```
+When we created the constructor above, we are assured that the object we create in our program will be automatically initialized. For instance, the declaration `integer v1;` automatically initialize our object `v1` and assigns the data members `a` and `b` the value  `2`. With normal member functions, we would be required to write a Y statement to invoke the constructor function and for each of the objects separately. So if we had a large number of objects, it would result to a lot of code which might be messy and difficult to read.
```

Comment:
the statement `we would be required to write a Y statement to invoke` seems a bit unclear. What is a `Y statement`
And the same sentence is a bit long. You could use an app like [Hemingway app](http://www.hemingwayapp.com/) to check for long sentences and find ways to shorten them. This is good to make your article an easy read. Try and get the lowest grade in Hemingway. The grades range from 10 to 1. The lower the grade, the better.